tests/test_models.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Model functions tests."""
+
+from __future__ import absolute_import, print_function
+
+import pytest
+from invenio_records.api import Record
+
+from invenio_communities.models import Community, InclusionRequest
+
+
+@pytest.mark.parametrize(('case_insensitive'), [
+    ('Beautiful'),
+    ('beautiful'),
+    ('bEAutiful'),
+    ('BEAUTIFUL'),
+])
+def test_filter_community(app, db, communities_for_filtering, user,
+                          case_insensitive):
+    """Test the community filter task."""
+
+    with app.app_context():
+        (comm0, comm1, comm2) = communities_for_filtering
+        rec1 = Record.create({
+            'title': 'Foobar and Bazbar',
+            'description': 'On Foobar, Bazbar and <b>more</b>.'
+        })
+        InclusionRequest.create(community=comm0, record=rec1, user=user)
+
+        # Case insensitive
+        def test_case_insensitivity(search_input):
+            results = Community.filter_communities(p=case_insensitive,
```

Comment:
Also since you wrap this test in a ``parametrize`` decorator, the whole test is repeated four times, hence all assertions below line 60 are also repeated four times (even though they don't depend on the ``case_insensitive`` parameter at all).